### PR TITLE
Implement getAddedChanges

### PR DESCRIPTION
**Pull request by vedantroy**
Implemented getAddedChanges which was in the JS backend, but not the Rust backend.

Not sure how to test this b/c many of the JS interop tests are failing right now due to  this (https://github.com/automerge/automerge-rs/commit/87410e4c8fe2a295c0c66e413f0451649193f541) I assume.

### CODE REVIEW

**Comment by jeffa5** [comment]
On that branch there is only currently one failing test.

**Comment by vedantroy** [comment]
Only 1 test failing now!

File: automerge-backend/src/backend.rs [hunk 1]
```diff
@@ -354,6 +354,25 @@ impl Backend {
             .and_then(|index| self.history.get(*index))
     }
 
+    pub fn get_changes_added<'a>(&self, other: &'a Self) -> Vec<&'a Change> {
+        let mut stack: Vec<_> = other.op_set.deps.iter().collect();
+        let mut seen_hashes = HashSet::new();
+        let mut added_change_hashes = Vec::new();
```
Comment (alexjg) [inline automerge-backend/src/backend.rs:360]: Do we need both `seen_hashes` and  `added_change_hashes`? They look like they are the same so we could just use the `seen_hashes`.

File: automerge-backend/src/backend.rs [hunk 2]
```diff
@@ -354,6 +354,25 @@ impl Backend {
             .and_then(|index| self.history.get(*index))
     }
 
+    pub fn get_changes_added<'a>(&self, other: &'a Self) -> Vec<&'a Change> {
```
Comment (alexjg) [inline automerge-backend/src/backend.rs:357]: Could we add some documentation on what this does? It's not immediately obvious to me from the name.

Comment (vedantroy) [reply automerge-backend/src/backend.rs:360]: Copied in the comments from the JS version.

File: automerge-backend/src/backend.rs (hunk 1 above)
Comment (vedantroy) [reply automerge-backend/src/backend.rs:365]: From looking at the comments in the JS version, it seems like we want to preserve the order of the DFS.

**Comment by vedantroy** [comment]
Also added a C binding, although it is untested because there are no tests for this method in the JS version of Automerge.

Will see if I can brainstorm a test.

**Pull request merged**